managed/devops/bin/yb_platform_util.py

```diff
@@ -0,0 +1,698 @@
+# Copyright (c) YugaByte, Inc.
+#
+# Copyright 2021 YugaByte, Inc. and Contributors
+#
+# Licensed under the Polyform Free Trial License 1.0.0 (the "License"); you
+# may not use this file except in compliance with the License. You
+# may obtain a copy of the License at
+#
+# https://github.com/YugaByte/yugabyte-db/blob/master/licenses/POLYFORM-FREE-TRIAL-LICENSE-1.0.0.txt
+
+import json
+import os
+import copy
+import collections
+import sys
+import argparse
+import textwrap
+import time
+from argparse import RawDescriptionHelpFormatter
+
+python_version = sys.version_info[0]
+if python_version == 2:
+    from urllib2 import HTTPError
+else:
+    from urllib.error import HTTPError
+
+SUCCESS = "success"
+FAIL = "failed"
+PROGRESS_BAR = 100
+script_path = os.path.dirname(os.path.abspath( __file__ ))
+
+
+def exception_handling(func):
+    def inner_function(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except HTTPError as e:
+            content = e.read().decode('utf-8')
+            if 'html>' in content:
```

Comment:
This is not a reliable way to detect errors. In the most general case, any response, even a JSON response, could contain the string `html>`. Is there a way to do this based on HTTP error codes? Or `<html` should be in the beginning of the content, or the content should not be valid JSON.

Reply:
Fixed